legacy/signals/signals.go

```diff
@@ -24,31 +24,68 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+var (
+	// ExitSignals are used to determine if an incoming os.Signal should cause termination.
+	ExitSignals map[os.Signal]bool = map[os.Signal]bool{
+		syscall.SIGHUP:  true,
+		syscall.SIGINT:  true,
+		syscall.SIGABRT: true,
+		syscall.SIGILL:  true,
+		syscall.SIGQUIT: true,
+		syscall.SIGTERM: true,
+		syscall.SIGSEGV: true,
+		syscall.SIGTSTP: true,
+	}
+	// ExitChannel is for gracefully terminating the LogSignals() function.
+	ExitChannel chan bool = make(chan bool, 1)
+	// SignalChannel is for passing observed OS signals.
+	SignalChannel chan os.Signal = make(chan os.Signal, 1)
+	// Debug is defined globally for mocking logrus.Debug statements.
+	Debug func(args ...interface{}) = logrus.Debug
+)
+
 // Watch concurrenlty logs debug statements when encountering interrupt signals from the OS.
 // This function relies on the os/signal package which may not capture SIGKILL and SIGSTOP.
 func Watch() {
-	c := make(chan os.Signal, 1)
+	Debug("Watching for OS Signals...")
 	// Observe all signals, excluding SIGKILL and SIGSTOP.
-	signal.Notify(c)
-	// Continuously watch for signals.
-	go func() {
-		logrus.Debug("Watching for OS Signals...")
-		for sig := range c {
-			logrus.Debug("Encoutered signal: ", sig.String())
-			// If the observed signal is a termination signal, we are
-			// expected to handle this by exiting the program. If we didn't
-			// exit, the program would only stop upon receiving a SIGKILL.
-			if sig == syscall.SIGHUP ||
-				sig == syscall.SIGINT ||
-				sig == syscall.SIGABRT ||
-				sig == syscall.SIGILL ||
-				sig == syscall.SIGQUIT ||
-				sig == syscall.SIGTERM ||
-				sig == syscall.SIGSEGV ||
-				sig == syscall.SIGTSTP {
-				logrus.Debug("Exiting from signal: ", sig.String())
-				os.Exit(0)
-			}
+	signal.Notify(SignalChannel)
```

Comment:
Nit: Looking at https://gobyexample.com/signals, apparently you can specify here which signals to listen for. So we don't even need ExitSignals.

Since you've got tests working and stuff, I'll chalk up the above to future refactoring work, but I wanted to capture the thought here.

Reply:
I talked about the trade offs between observing all signals and only some signals in this comment: https://github.com/kubernetes-sigs/k8s-container-image-promoter/pull/389#issuecomment-892299750
It seems that we may benifit from observing all signals, in an effort to improve the Auditor's visibility #392